libexec/rtld-elf/aarch64/reloc.c

```diff
@@ -428,8 +482,25 @@ reloc_iresolve_nonplt(Obj_Entry *obj, struct Struct_RtldLockState *lockstate)
 	obj->irelative_nonplt = false;
 	relalim = (const Elf_Rela *)((const char *)obj->rela + obj->relasize);
 	for (rela = obj->rela;  rela < relalim;  rela++) {
-		if (ELF_R_TYPE(rela->r_info) == R_AARCH64_IRELATIVE)
+		switch (ELF_R_TYPE(rela->r_info)) {
+		case R_MORELLO_IRELATIVE:
+#ifndef __CHERI_PURE_CAPABILITY__
+			_rtld_error("%s: R_MORELLO_IRELATIVE in hybrid binary",
```

Comment:
This is repeated quite a lot. Is it worth adding something like `REQUIRE_PURECAP(R_MORELLO_IRELATIVE)/REQUIRE_NON_PURECAP(R_MORELLO_IRELATIVE)`
that expands to 
```
_rtld_error("%s: " #reloc "in purecap/hybrid binary",
			    obj->path);
			return (-1);
```

Reply:
You still need the `#if(n)def` to avoid dead code and I don't like having return statements hidden in macros, so it'd just save on the error message printing which I don't think is worth it...

Reply:
Yes it's probably not worth it. However, we should probably add an #else to all these cases to remove the dead goto?

Reply:
Oops, yes, that was not intended

Reply:
I also inverted the conditions so the non-error case comes first